core/dbt/config/runtime.py

```diff
@@ -190,28 +201,52 @@ def _get_rendered_profile(
 
     @classmethod
     def collect_parts(cls: Type["RuntimeConfig"], args: Any) -> Tuple[Project, Profile]:
```

Comment:
The code in this function will be basically what the SB team will use to generate the profile and project, which can then be used with the `from_parts` method to generate Runtime configs.

The difference will be potentially overriding the profile name when it's collected like so:
```
profile = cls.collect_profile(args=args, profile_name="a_valid_profile_name")
```